[
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/285221709",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-285221709",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 285221709,
    "user": {
      "login": "kevinsawicki",
      "id": 671378,
      "avatar_url": "https://avatars0.githubusercontent.com/u/671378?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kevinsawicki",
      "html_url": "https://github.com/kevinsawicki",
      "followers_url": "https://api.github.com/users/kevinsawicki/followers",
      "following_url": "https://api.github.com/users/kevinsawicki/following{/other_user}",
      "gists_url": "https://api.github.com/users/kevinsawicki/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kevinsawicki/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kevinsawicki/subscriptions",
      "organizations_url": "https://api.github.com/users/kevinsawicki/orgs",
      "repos_url": "https://api.github.com/users/kevinsawicki/repos",
      "events_url": "https://api.github.com/users/kevinsawicki/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kevinsawicki/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2017-03-09T01:04:53Z",
    "updated_at": "2017-03-09T01:04:53Z",
    "body": "@gregor can you provide the full text of the stack trace?"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/286039585",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-286039585",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 286039585,
    "user": {
      "login": "Gloridea",
      "id": 322413,
      "avatar_url": "https://avatars2.githubusercontent.com/u/322413?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Gloridea",
      "html_url": "https://github.com/Gloridea",
      "followers_url": "https://api.github.com/users/Gloridea/followers",
      "following_url": "https://api.github.com/users/Gloridea/following{/other_user}",
      "gists_url": "https://api.github.com/users/Gloridea/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Gloridea/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Gloridea/subscriptions",
      "organizations_url": "https://api.github.com/users/Gloridea/orgs",
      "repos_url": "https://api.github.com/users/Gloridea/repos",
      "events_url": "https://api.github.com/users/Gloridea/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Gloridea/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-13T08:09:36Z",
    "updated_at": "2017-03-13T08:09:36Z",
    "body": "@kevinsawicki  I have a same issue. full stack trace is:\r\n\r\n```\r\nError: Cannot find module '../screen'\r\n    at Module._resolveFilename (module.js:470:15)\r\n    at Function.Module._resolveFilename (/Users/gloridea/Workspaces/myapp/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)\r\n    at Function.Module._load (module.js:418:25)\r\n    at Module.require (module.js:498:17)\r\n    at require (internal/module.js:20:19)\r\n    at Object.get [as screen] (/Users/gloridea/Workspaces/myapp/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/browser/api/exports/electron.js:91:14)\r\n    at App.<anonymous> (/Users/gloridea/Workspaces/myapp/src/electron/main.ts:111:18)\r\n    at Generator.next (<anonymous>)\r\n    at /Users/gloridea/Workspaces/myapp/output/electron/main.js:7:71\r\n    at __awaiter (/Users/gloridea/Workspaces/myapp/output/electron/main.js:3:12)\r\n    at App.<anonymous> (/Users/gloridea/Workspaces/myapp/output/electron/main.js:99:12)\r\n    at emitTwo (events.js:111:20)\r\n    at App.emit (events.js:191:7) code: 'MODULE_NOT_FOUND' }\r\n(node:52416) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module '../screen'\r\n(node:52416) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.\r\n```\r\n\r\nI also tried to check followings:\r\n```\r\nconsole.log('electron_1', Object.keys(electron_1));\r\n```\r\n\r\nand it's result:\r\n```\r\nelectron_1 [ 'clipboard',\r\n  'crashReporter',\r\n  'nativeImage',\r\n  'shell',\r\n  'app',\r\n  'autoUpdater',\r\n  'BrowserWindow',\r\n  'contentTracing',\r\n  'dialog',\r\n  'globalShortcut',\r\n  'ipcMain',\r\n  'Menu',\r\n  'MenuItem',\r\n  'net',\r\n  'powerMonitor',\r\n  'powerSaveBlocker',\r\n  'protocol',\r\n  'screen',\r\n  'session',\r\n  'systemPreferences',\r\n  'Tray',\r\n  'webContents' ]\r\n```"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/286466350",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-286466350",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 286466350,
    "user": {
      "login": "gregor",
      "id": 10243309,
      "avatar_url": "https://avatars2.githubusercontent.com/u/10243309?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gregor",
      "html_url": "https://github.com/gregor",
      "followers_url": "https://api.github.com/users/gregor/followers",
      "following_url": "https://api.github.com/users/gregor/following{/other_user}",
      "gists_url": "https://api.github.com/users/gregor/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gregor/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gregor/subscriptions",
      "organizations_url": "https://api.github.com/users/gregor/orgs",
      "repos_url": "https://api.github.com/users/gregor/repos",
      "events_url": "https://api.github.com/users/gregor/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gregor/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-14T15:54:16Z",
    "updated_at": "2017-03-14T15:54:16Z",
    "body": "@Gloridea Thanks for adding this. Didn't find the time to do so myself yet."
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/287220068",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-287220068",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 287220068,
    "user": {
      "login": "kevinsawicki",
      "id": 671378,
      "avatar_url": "https://avatars0.githubusercontent.com/u/671378?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kevinsawicki",
      "html_url": "https://github.com/kevinsawicki",
      "followers_url": "https://api.github.com/users/kevinsawicki/followers",
      "following_url": "https://api.github.com/users/kevinsawicki/following{/other_user}",
      "gists_url": "https://api.github.com/users/kevinsawicki/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kevinsawicki/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kevinsawicki/subscriptions",
      "organizations_url": "https://api.github.com/users/kevinsawicki/orgs",
      "repos_url": "https://api.github.com/users/kevinsawicki/repos",
      "events_url": "https://api.github.com/users/kevinsawicki/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kevinsawicki/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2017-03-16T23:10:31Z",
    "updated_at": "2017-03-16T23:10:31Z",
    "body": "@Gloridea are you using `app.makeSingleInstance()` as well?"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/287227333",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-287227333",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 287227333,
    "user": {
      "login": "kevinsawicki",
      "id": 671378,
      "avatar_url": "https://avatars0.githubusercontent.com/u/671378?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kevinsawicki",
      "html_url": "https://github.com/kevinsawicki",
      "followers_url": "https://api.github.com/users/kevinsawicki/followers",
      "following_url": "https://api.github.com/users/kevinsawicki/following{/other_user}",
      "gists_url": "https://api.github.com/users/kevinsawicki/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kevinsawicki/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kevinsawicki/subscriptions",
      "organizations_url": "https://api.github.com/users/kevinsawicki/orgs",
      "repos_url": "https://api.github.com/users/kevinsawicki/repos",
      "events_url": "https://api.github.com/users/kevinsawicki/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kevinsawicki/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2017-03-16T23:53:05Z",
    "updated_at": "2017-03-16T23:53:05Z",
    "body": "@Gloridea also, are you able to provide an example app that reproduces this for you?"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/288001259",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-288001259",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 288001259,
    "user": {
      "login": "Gloridea",
      "id": 322413,
      "avatar_url": "https://avatars2.githubusercontent.com/u/322413?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Gloridea",
      "html_url": "https://github.com/Gloridea",
      "followers_url": "https://api.github.com/users/Gloridea/followers",
      "following_url": "https://api.github.com/users/Gloridea/following{/other_user}",
      "gists_url": "https://api.github.com/users/Gloridea/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Gloridea/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Gloridea/subscriptions",
      "organizations_url": "https://api.github.com/users/Gloridea/orgs",
      "repos_url": "https://api.github.com/users/Gloridea/repos",
      "events_url": "https://api.github.com/users/Gloridea/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Gloridea/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-21T07:51:37Z",
    "updated_at": "2017-03-21T08:02:51Z",
    "body": "@kevinsawicki \r\nSorry for may late answer. \r\n1. Yes, I'm using `app.makeSingleInstance()`.\r\n2. I'll have a try to create sample error app soon.\r\n\r\n"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/290695900",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-290695900",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 290695900,
    "user": {
      "login": "oukan",
      "id": 1324257,
      "avatar_url": "https://avatars1.githubusercontent.com/u/1324257?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/oukan",
      "html_url": "https://github.com/oukan",
      "followers_url": "https://api.github.com/users/oukan/followers",
      "following_url": "https://api.github.com/users/oukan/following{/other_user}",
      "gists_url": "https://api.github.com/users/oukan/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/oukan/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/oukan/subscriptions",
      "organizations_url": "https://api.github.com/users/oukan/orgs",
      "repos_url": "https://api.github.com/users/oukan/repos",
      "events_url": "https://api.github.com/users/oukan/events{/privacy}",
      "received_events_url": "https://api.github.com/users/oukan/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-31T12:12:57Z",
    "updated_at": "2017-03-31T12:12:57Z",
    "body": "Electron version: at least 1.6.2\r\nOperating system: Windows 7\r\nAlso have this problem"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291190192",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291190192",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291190192,
    "user": {
      "login": "arantes555",
      "id": 99662,
      "avatar_url": "https://avatars2.githubusercontent.com/u/99662?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/arantes555",
      "html_url": "https://github.com/arantes555",
      "followers_url": "https://api.github.com/users/arantes555/followers",
      "following_url": "https://api.github.com/users/arantes555/following{/other_user}",
      "gists_url": "https://api.github.com/users/arantes555/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/arantes555/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/arantes555/subscriptions",
      "organizations_url": "https://api.github.com/users/arantes555/orgs",
      "repos_url": "https://api.github.com/users/arantes555/repos",
      "events_url": "https://api.github.com/users/arantes555/events{/privacy}",
      "received_events_url": "https://api.github.com/users/arantes555/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-03T16:06:53Z",
    "updated_at": "2017-04-03T16:17:58Z",
    "body": "Seeing the same issue here, on electron 1.6.2, on a app that does use makeSingleInstance, but only from one of our users. More precisely, what we get is an error on `../net`.\r\nFull error stack (not very informative):\r\n```\r\nUNKNOWNMODULE_NOT_FOUND Cannot find module '../net'\r\n    at APIError.t (C:\\Users\\XXXX\\AppData\\Local\\Programs\\Seald\\resources\\app.asar\\background.js:1:545)\r\n    at APIError (C:\\Users\\XXXX\\AppData\\Local\\Programs\\Seald\\resources\\app.asar\\background.js:1:22793)\r\n    at C:\\Users\\XXXX\\AppData\\Local\\Programs\\Seald\\resources\\app.asar\\background.js:1:23984\r\n    at Generator.throw ()\r\n    at r (C:\\Users\\XXXX\\AppData\\Local\\Programs\\Seald\\resources\\app.asar\\background.js:1:5017)\r\n    at C:\\Users\\XXXX\\AppData\\Local\\Programs\\Seald\\resources\\app.asar\\background.js:1:5136\r\n```\r\n\r\n( @gregor : interestingly, it's also an encryption related app that we are seeing the problem on :) Coincidence?)"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291198043",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291198043",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291198043,
    "user": {
      "login": "tex0l",
      "id": 7548264,
      "avatar_url": "https://avatars2.githubusercontent.com/u/7548264?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/tex0l",
      "html_url": "https://github.com/tex0l",
      "followers_url": "https://api.github.com/users/tex0l/followers",
      "following_url": "https://api.github.com/users/tex0l/following{/other_user}",
      "gists_url": "https://api.github.com/users/tex0l/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/tex0l/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/tex0l/subscriptions",
      "organizations_url": "https://api.github.com/users/tex0l/orgs",
      "repos_url": "https://api.github.com/users/tex0l/repos",
      "events_url": "https://api.github.com/users/tex0l/events{/privacy}",
      "received_events_url": "https://api.github.com/users/tex0l/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-03T16:33:31Z",
    "updated_at": "2017-04-03T16:33:31Z",
    "body": "Just completing what @arantes555 has provided, it happened on Win7-64."
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291296682",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291296682",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291296682,
    "user": {
      "login": "kevinsawicki",
      "id": 671378,
      "avatar_url": "https://avatars0.githubusercontent.com/u/671378?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kevinsawicki",
      "html_url": "https://github.com/kevinsawicki",
      "followers_url": "https://api.github.com/users/kevinsawicki/followers",
      "following_url": "https://api.github.com/users/kevinsawicki/following{/other_user}",
      "gists_url": "https://api.github.com/users/kevinsawicki/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kevinsawicki/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kevinsawicki/subscriptions",
      "organizations_url": "https://api.github.com/users/kevinsawicki/orgs",
      "repos_url": "https://api.github.com/users/kevinsawicki/repos",
      "events_url": "https://api.github.com/users/kevinsawicki/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kevinsawicki/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2017-04-03T22:47:08Z",
    "updated_at": "2017-04-03T22:47:08Z",
    "body": "I'm having trouble reproducing this issue exactly on Windows 10 but am seeing one major difference between 1.4.15 and 1.6.5, the `app` `ready` event fires for each instance that is created in 1.6.x while in 1.4.x it is only fired for the first app instance.\r\n\r\nAre people who are experiencing this issue listening in their apps to the ready event even when `app.makeSingleInstance` returns `false`?\r\n\r\nThis was the app I tried with:\r\n\r\n```js\r\nconst {app, BrowserWindow} = require('electron')\r\nlet window = null\r\n\r\nconst shouldQuit = app.makeSingleInstance(() => {\r\n  window.focus()\r\n  console.log('app.makeSingleInstance')\r\n  console.log(require('electron').screen.getCursorScreenPoint())\r\n})\r\n\r\nif (shouldQuit) {\r\n  app.quit()\r\n} else {\r\n  app.on('ready', () => {\r\n    window = new BrowserWindow()\r\n    window.loadURL('about:blank')\r\n  })\r\n}\r\n```\r\n\r\nBut if I switch it to the following I do see multiple dialog boxes in 1.6.x but not in 1.4.x:\r\n\r\n```js\r\nconst {app, dialog, BrowserWindow} = require('electron')\r\nlet window = null\r\n\r\nconst shouldQuit = app.makeSingleInstance(() => {\r\n  window.focus()\r\n  console.log('app.makeSingleInstance')\r\n  console.log(require('electron').screen.getCursorScreenPoint())\r\n})\r\n\r\nif (shouldQuit) {\r\n  app.quit()\r\n}\r\n\r\napp.on('ready', () => {\r\n  dialog.showErrorBox('app', 'app is ready')\r\n  window = new BrowserWindow()\r\n  window.loadURL('about:blank')\r\n})\r\n```"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291947839",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291947839",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291947839,
    "user": {
      "login": "Arm1stice",
      "id": 6374653,
      "avatar_url": "https://avatars2.githubusercontent.com/u/6374653?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Arm1stice",
      "html_url": "https://github.com/Arm1stice",
      "followers_url": "https://api.github.com/users/Arm1stice/followers",
      "following_url": "https://api.github.com/users/Arm1stice/following{/other_user}",
      "gists_url": "https://api.github.com/users/Arm1stice/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Arm1stice/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Arm1stice/subscriptions",
      "organizations_url": "https://api.github.com/users/Arm1stice/orgs",
      "repos_url": "https://api.github.com/users/Arm1stice/repos",
      "events_url": "https://api.github.com/users/Arm1stice/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Arm1stice/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-05T18:10:42Z",
    "updated_at": "2017-04-05T18:10:42Z",
    "body": "I've been able to reproduce the bug on macOS Sierra"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291948946",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291948946",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291948946,
    "user": {
      "login": "kevinsawicki",
      "id": 671378,
      "avatar_url": "https://avatars0.githubusercontent.com/u/671378?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kevinsawicki",
      "html_url": "https://github.com/kevinsawicki",
      "followers_url": "https://api.github.com/users/kevinsawicki/followers",
      "following_url": "https://api.github.com/users/kevinsawicki/following{/other_user}",
      "gists_url": "https://api.github.com/users/kevinsawicki/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kevinsawicki/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kevinsawicki/subscriptions",
      "organizations_url": "https://api.github.com/users/kevinsawicki/orgs",
      "repos_url": "https://api.github.com/users/kevinsawicki/repos",
      "events_url": "https://api.github.com/users/kevinsawicki/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kevinsawicki/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2017-04-05T18:14:31Z",
    "updated_at": "2017-04-05T18:14:31Z",
    "body": "@Arm1stice are you able to provide a reproduction app that demonstrates it? Thanks."
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291949621",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291949621",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291949621,
    "user": {
      "login": "Arm1stice",
      "id": 6374653,
      "avatar_url": "https://avatars2.githubusercontent.com/u/6374653?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Arm1stice",
      "html_url": "https://github.com/Arm1stice",
      "followers_url": "https://api.github.com/users/Arm1stice/followers",
      "following_url": "https://api.github.com/users/Arm1stice/following{/other_user}",
      "gists_url": "https://api.github.com/users/Arm1stice/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Arm1stice/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Arm1stice/subscriptions",
      "organizations_url": "https://api.github.com/users/Arm1stice/orgs",
      "repos_url": "https://api.github.com/users/Arm1stice/repos",
      "events_url": "https://api.github.com/users/Arm1stice/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Arm1stice/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-05T18:16:47Z",
    "updated_at": "2017-04-05T18:16:47Z",
    "body": "@kevinsawicki My laptop is dying on me, but I'll post a link to my repo when I get home. I would like to note, though, my error was reproduced without using app.makeSingleInstance()"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291978656",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291978656",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291978656,
    "user": {
      "login": "Arm1stice",
      "id": 6374653,
      "avatar_url": "https://avatars2.githubusercontent.com/u/6374653?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Arm1stice",
      "html_url": "https://github.com/Arm1stice",
      "followers_url": "https://api.github.com/users/Arm1stice/followers",
      "following_url": "https://api.github.com/users/Arm1stice/following{/other_user}",
      "gists_url": "https://api.github.com/users/Arm1stice/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Arm1stice/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Arm1stice/subscriptions",
      "organizations_url": "https://api.github.com/users/Arm1stice/orgs",
      "repos_url": "https://api.github.com/users/Arm1stice/repos",
      "events_url": "https://api.github.com/users/Arm1stice/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Arm1stice/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-05T19:56:34Z",
    "updated_at": "2017-04-05T19:59:11Z",
    "body": "@kevinsawicki \r\nhttps://gist.github.com/Arm1stice/9e3c99d6d4353dafd2ba9f9e3ec94db9\r\n\r\nRunning this gist with electron will produce the error. My theory is that, because the last open window closes before the next code runs, electron possibly attempts to shut down."
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/291979237",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-291979237",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 291979237,
    "user": {
      "login": "Arm1stice",
      "id": 6374653,
      "avatar_url": "https://avatars2.githubusercontent.com/u/6374653?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Arm1stice",
      "html_url": "https://github.com/Arm1stice",
      "followers_url": "https://api.github.com/users/Arm1stice/followers",
      "following_url": "https://api.github.com/users/Arm1stice/following{/other_user}",
      "gists_url": "https://api.github.com/users/Arm1stice/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Arm1stice/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Arm1stice/subscriptions",
      "organizations_url": "https://api.github.com/users/Arm1stice/orgs",
      "repos_url": "https://api.github.com/users/Arm1stice/repos",
      "events_url": "https://api.github.com/users/Arm1stice/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Arm1stice/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-05T19:58:40Z",
    "updated_at": "2017-04-05T19:58:40Z",
    "body": "Is there the possibility that this only happens with some users as @arantes555 suggests because electron could be shutting down faster on better computers?"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/292073636",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-292073636",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 292073636,
    "user": {
      "login": "tex0l",
      "id": 7548264,
      "avatar_url": "https://avatars2.githubusercontent.com/u/7548264?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/tex0l",
      "html_url": "https://github.com/tex0l",
      "followers_url": "https://api.github.com/users/tex0l/followers",
      "following_url": "https://api.github.com/users/tex0l/following{/other_user}",
      "gists_url": "https://api.github.com/users/tex0l/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/tex0l/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/tex0l/subscriptions",
      "organizations_url": "https://api.github.com/users/tex0l/orgs",
      "repos_url": "https://api.github.com/users/tex0l/repos",
      "events_url": "https://api.github.com/users/tex0l/events{/privacy}",
      "received_events_url": "https://api.github.com/users/tex0l/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-06T05:36:10Z",
    "updated_at": "2017-04-06T05:36:10Z",
    "body": "@kevinsawicki in the case of @arantes555 (and mine), I've checked, yes we do listen for the `'ready'` event even if `app.makeSingleInstance` returns `false` (it's a bit dirty though)."
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/294303518",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-294303518",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 294303518,
    "user": {
      "login": "kivS",
      "id": 3810567,
      "avatar_url": "https://avatars0.githubusercontent.com/u/3810567?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/kivS",
      "html_url": "https://github.com/kivS",
      "followers_url": "https://api.github.com/users/kivS/followers",
      "following_url": "https://api.github.com/users/kivS/following{/other_user}",
      "gists_url": "https://api.github.com/users/kivS/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/kivS/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/kivS/subscriptions",
      "organizations_url": "https://api.github.com/users/kivS/orgs",
      "repos_url": "https://api.github.com/users/kivS/repos",
      "events_url": "https://api.github.com/users/kivS/events{/privacy}",
      "received_events_url": "https://api.github.com/users/kivS/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-04-15T16:27:49Z",
    "updated_at": "2017-04-15T16:27:49Z",
    "body": "Had the same problem using makeSingleInstance, I solved it using app.exit() instead of app.quit().\r\n\r\n```javascript\r\nconst shouldQuit = app.makeSingleInstance(() =>{\r\n  if(mainWindow) mainWindow.show();\r\n})\r\nif(shouldQuit) app.exit();\r\n```\r\nsomething to do with app.exit not having to emit events before quitting "
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/309130755",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-309130755",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 309130755,
    "user": {
      "login": "gch1p",
      "id": 18737999,
      "avatar_url": "https://avatars2.githubusercontent.com/u/18737999?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gch1p",
      "html_url": "https://github.com/gch1p",
      "followers_url": "https://api.github.com/users/gch1p/followers",
      "following_url": "https://api.github.com/users/gch1p/following{/other_user}",
      "gists_url": "https://api.github.com/users/gch1p/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gch1p/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gch1p/subscriptions",
      "organizations_url": "https://api.github.com/users/gch1p/orgs",
      "repos_url": "https://api.github.com/users/gch1p/repos",
      "events_url": "https://api.github.com/users/gch1p/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gch1p/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-06-16T20:49:25Z",
    "updated_at": "2017-06-16T20:49:35Z",
    "body": "On windows, I can sometimes catch this bug just by running my app from command line. I'm not sure how to reproduce it or why it happens, but it happened to me twice in the last hour. Both times I was launching the app from cmd.exe"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/309140576",
    "html_url": "https://github.com/electron/electron/issues/8862#issuecomment-309140576",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/8862",
    "id": 309140576,
    "user": {
      "login": "gch1p",
      "id": 18737999,
      "avatar_url": "https://avatars2.githubusercontent.com/u/18737999?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gch1p",
      "html_url": "https://github.com/gch1p",
      "followers_url": "https://api.github.com/users/gch1p/followers",
      "following_url": "https://api.github.com/users/gch1p/following{/other_user}",
      "gists_url": "https://api.github.com/users/gch1p/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gch1p/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gch1p/subscriptions",
      "organizations_url": "https://api.github.com/users/gch1p/orgs",
      "repos_url": "https://api.github.com/users/gch1p/repos",
      "events_url": "https://api.github.com/users/gch1p/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gch1p/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-06-16T21:41:13Z",
    "updated_at": "2017-06-16T21:41:13Z",
    "body": "I figured it: it's because of `process.noAsar = true`, but I don't think it's a feature, right?"
  }
]